### PR TITLE
polygons should have 4+ points #883

### PR DESCRIPTION
**Pull request by akrherz**

### CODE REVIEW

File: lib/cartopy/mpl/patch.py [hunk 1]
```diff
@@ -185,13 +185,10 @@ def path_to_geos(path, force_ccw=False):
                                      axis=1)
         if all(verts_same_as_first):
             geom = sgeom.Point(path_verts[0, :])
-        elif (path_verts.shape[0] > 2 and
+        elif (path_verts.shape[0] > 3 and
                 (path_codes[-1] == Path.CLOSEPOLY or
                  verts_same_as_first[-1])):
-            if path_codes[-1] == Path.CLOSEPOLY:
```
Comment (QuLogic) [inline lib/cartopy/mpl/patch.py]: For `CLOSEPOLY`, the last location should be ignored; see [Matplotlib's Path tutorial](https://matplotlib.org/users/path_tutorial.html). Thus, I don't think this condition should be removed, unless you can guarantee that the last point is at the first point (which seems unlikely given the `or verts_same_as_first[-1]` bit.)

**Comment by QuLogic** [comment]
Looks reasonable, except for the comment above. I believe you still need to sign the CLA though.

**Comment by akrherz** [comment]
Thanks @QuLogic , your suggestion is safer than my assumption that the effectively ignored last point associated with `Path.CLOSEPOLY` may not always equal the first point.  Hopefully properly signed the CLA as well.

**Comment by ajdawson** [comment]
@marqh - please confirm when a CLA from @akrherz has been processed, thanks.

**Comment by akrherz** [comment]
@ajdawson I thought signing the CLA involved adding my name to `docs/source/contributors.rst`, which I did in the commit above.  Otherwise, what needs to be done?

**Comment by ajdawson** [comment]
@akrherz - You need to fill in the [CLA form](http://scitools.org.uk/documents/cla_form.pdf) and the Met Office will process it, shouldn't take long. If you are unfamiliar the the CLA concept see here: http://oss-watch.ac.uk/resources/cla

**Comment by lbdreyer** [comment]
We have received the cla from @akrherz so this is now ready to go in (also see https://github.com/SciTools/scitools.org.uk/pull/155)

**Pull request merged**